compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -5084,11 +5086,9 @@ private ParserRuleContext getNextRuleForVarName() {
     private ParserRuleContext getNextRuleForSemicolon(int nextLookahead) {
         STToken nextToken;
         ParserRuleContext parentCtx = getParentContext();
-        if (parentCtx == ParserRuleContext.EXTERNAL_FUNC_BODY) {
-            endContext(); // end external func-body
-            return getNextRuleForSemicolon(nextLookahead);
-        } else if (parentCtx == ParserRuleContext.QUERY_EXPRESSION) {
-            endContext(); // end expression
+        if (parentCtx == ParserRuleContext.EXTERNAL_FUNC_BODY || parentCtx == ParserRuleContext.ON_FAIL_CHECK
+                || parentCtx == ParserRuleContext.QUERY_EXPRESSION) {
+            endContext();
```

Comment:
Do we need these changes? Since we already end the context for `ParserRuleContext.ON_FAIL_CHECK` at `ON_FAIL_CHECK_END` we should not reach here with a `ON_FAIL_CHECK` as the parentCtx

Reply:
No, I will update.